crates/bevy_asset/src/handle.rs

```diff
@@ -97,6 +97,7 @@ impl HandleId {
 ///
 #[derive(Component, Reflect)]
 #[reflect(Component, Default)]
+#[reflect(custom_where(T: Asset))]
 pub struct Handle<T>
 where
     T: Asset,
```

Comment:
A little, but they serve two separate purposes. A `where` like this defines bounds for the entire type itself. The `custom_where` provides bounds only for the generated reflection impls.

If we want to avoid this apparent redundancy, we could consider renaming the attribute to something else.